rust/operator-binary/src/controller.rs

```diff
@@ -104,25 +106,26 @@ pub enum Error {
     ProductConfigLoadFailed,
     #[snafu(display("failed to create rego rules for authorization"))]
     RegoRuleAuthorizationError { source: authorization::Error },
-    #[snafu(display("failed to get config map [{}/{}]", namespace, name,))]
+    #[snafu(display("failed to get config map {}", config_map))]
     MissingConfigMap {
         source: stackable_operator::error::Error,
-        name: String,
-        namespace: String,
+        config_map: ObjectRef<ConfigMap>,
     },
     #[snafu(display(
-        "failed to get [{}] connection string from config map [{}/{}]",
+        "failed to get [{}] connection string from config map {}",
         product,
-        namespace,
-        name
+        config_map
     ))]
     MissingConnectString {
         product: String,
-        name: String,
-        namespace: String,
+        config_map: ObjectRef<ConfigMap>,
     },
     #[snafu(display("failed to processing authentication config element from k8s"))]
     FailedProcessingAuthentication { source: authentication::Error },
+    #[snafu(display("internal operator failure"))]
+    InternalOperatorFailure { source: stackable_trino_crd::Error },
```

Comment:
I'd rather not have this kind of generic error message, even if it could truly never be caused by downstream users something more descriptive would probably be helpful for getting to the bottom of it.

Then again, in this specific case it's really a case of product-config suckage (ideally, we'd just thread through `TrinoRole` rather than doing the song and dance converting it between `TrinoRole` and `String`).

Reply:
Yeah i could not come up with a better name unfortunately.